openedx_tagging/core/tagging/rest_api/v1/views.py

```diff
@@ -0,0 +1,180 @@
+        {
+            "name": "Taxonomy New Name",
+            "description": "This is a new description",
+            "enabled": False,
+            "required": False,
+            "allow_multiple": False,
+            "allow_free_text": True,
+            "taxonomy_class": "class", #ToDo: add a example here
+        }
+        PATCH api/tagging/v1/taxonomy/:id                                          - Partially update a taxonomy
+        {
+            "name": "Taxonomy New Name",
+        }
+
+    **Update Query Returns**
+        * 200 - Success
+        * 403 - Permission denied
+
+    **Delete Parameters**
+        * id (required): - The id of the taxonomy to delete
+
+    **Delete Example Requests**
+        DELETE api/tagging/v1/taxonomy/:id                                         - Delete a taxonomy
+
+    **Delete Query Returns**
+        * 200 - Success
+        * 404 - Taxonomy not found
+        * 403 - Permission denied
+
+    """
+
+
+    serializer_class = TaxonomySerializer
+    lookup_field = "id"
+    permission_classes = [TaxonomyObjectPermissions]
+
+    def get_object(self):
+        """
+        Return the requested taxonomy object, if the user has appropriate
+        permissions.
+        """
+        id = self.kwargs.get("id")
+        taxonomy = get_taxonomy(id)
+        if not taxonomy:
+            raise Http404("Taxonomy not found")
+        self.check_object_permissions(self.request, taxonomy)
+
+        return taxonomy
+
+    def get_queryset(self):
+        """
+        Return a list of taxonomies. If you want the disabled taxonomies, pass enabled=False. If you want all taxonomies (both enabled and disabled), pass enabled=None.
+        """
+        query_params = TaxonomyListQueryParamsSerializer(
+            data=self.request.query_params.dict()
+        )
+        query_params.is_valid(raise_exception=True)
+        enabled = query_params.data.get("enabled", None)
+
+        return get_taxonomies(enabled)
+
+    def create(self, request):
+        """
+        Create a new taxonomy.
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        taxonomy = create_taxonomy(**serializer.validated_data)
+        return Response(
+            self.serializer_class(taxonomy).data, status=status.HTTP_201_CREATED
+        )
+
+    def update(self, request, *args, **kwargs):
+        """
+        Update a taxonomy.
+        """
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        update_taxonomy(instance.id, **serializer.validated_data)
+
+        return Response(serializer.data)
+
+    def perform_destroy(self, instance):
+        """
+        Delete a taxonomy.
+        """
+        delete_taxonomy(instance.id)
```

Comment:
I don't think these methods should be re-implemented. We should use `api.create_taxonomy` when creating, but it's ok to let the default ModelViewSet handle updates and destroy, no need to create or use API methods for update/delete.

Also: it's better to override just `perform_create` instead of the whole `create` method.

```suggestion
    def perform_create(self, serializer):
        """
        Create a new taxonomy.
        """
        serializer.instance = create_taxonomy(**serializer.validated_data)
```